server/src/main/kotlin/org/itmodreamteam/myrest/server/controller/UserController.kt

```diff
@@ -0,0 +1,33 @@
+package org.itmodreamteam.myrest.server.controller
+
+import org.itmodreamteam.myrest.server.security.UserAuthentication
+import org.itmodreamteam.myrest.server.service.user.UserService
+import org.itmodreamteam.myrest.shared.user.*
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/users")
+class UserController(private val userService: UserService) {
+
+    @GetMapping("/me")
```

Comment:
/users/{id} - is common endpoint to fetch user info of a specific user.
/users/me may interfere with a user with id equals to me. It's better to distinguish such "special" endpoints by using special characters, e.g. `@`: /users/@me

Reply:
Why is 4 better than 1-3?
1. We specify that user ID is a number of long type, therefore `me` never collides with user ID (current case).
2. We specify that user ID is a UUID, therefore `me` never collides with user ID.
2. We specify that user ID is a String that contains at least 16 characters, therefore `me` never collides with user ID.
3. We specify that user ID is a String that never starts with @, therefore `@me` never collides with user ID.

Reply:
All is actually fine. Let's stick with 1.